core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java

```diff
@@ -1680,6 +1688,61 @@ private ObjList<Function> generateCastFunctions(
         return castFunctions;
     }
 
+    private RecordCursorFactory generateFill(QueryModel model, RecordCursorFactory groupByFactory, SqlExecutionContext executionContext) throws SqlException {
+        if (!isFillStridePresent(model.getNestedModel())) {
+            return groupByFactory;
+        } else {
+            try {
+                final QueryModel nested = model.getNestedModel();
+                Function fillFromFunc = TimestampConstant.NULL;
+                Function fillToFunc = TimestampConstant.NULL;
+                final ExpressionNode fillFrom = nested.getFillFrom();
+                final ExpressionNode fillTo = nested.getFillTo();
+                final ExpressionNode fillStride = nested.getFillStride();
+
+                ObjList<ExpressionNode> fillValuesExprs = nested.getFillValue();
+                ObjList<Function> fillValues = new ObjList<>(fillValuesExprs.size());
+
+                ExpressionNode expr;
+                for (int i = 0, n = fillValuesExprs.size(); i < n; i++) {
+                    expr = fillValuesExprs.getQuick(0);
+                    if (isNoneKeyword(expr.token)) {
+                        return groupByFactory;
+                    }
+                    final Function fillValueFunc = functionParser.parseFunction(expr, EmptyRecordMetadata.INSTANCE, executionContext);
+                    fillValues.add(fillValueFunc);
+                }
+
+                if (fillValues.size() == 0) {
+                    return groupByFactory;
+                }
+
+                if (fillValues.size() == 1 && isNoneKeyword(fillValues.getQuick(0).getName())) {
+                    return groupByFactory;
+                }
+
+                if (fillFrom != null) {
+                    fillFromFunc = functionParser.parseFunction(fillFrom, EmptyRecordMetadata.INSTANCE, executionContext);
+                    coerceRuntimeConstantType(fillFromFunc, ColumnType.TIMESTAMP, executionContext, "from lower bound must be a constant expression convertible to a TIMESTAMP", -1);
+                }
+
+                if (fillTo != null) {
+                    fillToFunc = functionParser.parseFunction(fillTo, EmptyRecordMetadata.INSTANCE, executionContext);
+                    coerceRuntimeConstantType(fillToFunc, ColumnType.TIMESTAMP, executionContext, "to upper bound must be a constant expression convertible to a TIMESTAMP", -1);
+                }
+
+                fillFromFunc.init(null, executionContext);
+                fillToFunc.init(null, executionContext);
+
+
+                return new FillRangeRecordCursorFactory(groupByFactory.getMetadata(), groupByFactory, fillFromFunc, fillToFunc, fillStride.token, fillValues, getTimestampIndex(nested, groupByFactory.getMetadata()));
+            } catch (Throwable e) {
+                Misc.free(groupByFactory);
```

Comment:
It might be a good idea to release `fillValues`, `fillFromFunc`, `fillToFunc` here too if any of them were generated.

Reply:
These are very fragile assumptions. E.g. if one day we decide to store constants/bind variable values off-heap, it'll break unnoticeably. It's important to do our best when it comes to freeing resources in case of an error.

Reply:
I can push the change, no prob. Will do it tomorrow morning, if that's ok. WDYT?

Reply:
Pushed in 2927fe4c70bc107c49b7d880a6fb46e03aed02a3